simulators/ethereum/engine/suites/cancun/steps.go

```diff
@@ -27,6 +27,7 @@ import (
 type CancunTestContext struct {
```

Comment:
Could we take this opportunity to rename this to something more generic, like `TestContext`.

And also create a `func NewTestContext() *TestContext {` that performs all init operations required such as the `t.DevP2PConnections = make(map[uint64]*devp2p.Conn)` that we do in line 782.

Reply:
We could also create a `func (t *TestContext) Close() error {` that we could `defer t.Close()` after we call `NewTestContext()` when creating the context to do some proper cleanup such as the closing of the connections that are still open in `t.DevP2PConnections`.

Reply:
Very happy to extend to this!

My only rebutle is that we'd be adding the extra initialization steps for all the CancunBaseSpec tests, where its only required for 2 of them.

Maybe we have a generic `TestContext` and additionally a specific `DevP2PTestContext`, the latter having the init steps you mentioned. Where we only init these steps for the DevP2P tests?

Reply:
I added you suggestions in: [25ffdf9](https://github.com/ethereum/hive/pull/1052/commits/25ffdf9e225b550f693787a07a680e25ff0c256f)

Let me know what you think :)